audio_features.py

```diff
@@ -0,0 +1,176 @@
+
+
+def get_audio_features(uri):
+    """Posts audio features to the database
+    :param uri: uri
+    :return: -
+    """
+    try:
+        uri = str(uri)
+        res = re.findall(r':(?: *([\w.-]+):)', uri)
+        str_res = ' '.join([str(word) for word in res])
+
+        if str_res in ['playlist', 'userplaylist']:
+            # from the playlist get URIs for each artist
+            artist_uris_total = get_artists_from(uri)
+            # from artist uris get a list of album uris
+            albums_uris_total = []
+            for artist_uri in artist_uris_total:
+                album_uris = get_albums_from(artist_uri)
+                albums_uris_total.extend(album_uris)
+            # from a list of albums get tracks
+            track_uris_total = []
+            for albums_uri in albums_uris_total:
+                tracks_uris = get_tracks_from(albums_uri)
+                track_uris_total.extend(tracks_uris)
+            print(track_uris_total)
+            for track_uri in track_uris_total:
+                features_to_db(track_uri)
+
+        elif str_res == 'artist':
+            albums_uris_total = get_albums_from(uri)
+            track_uris_total = []
+            for albums_uri in albums_uris_total:
+                tracks_uris = get_tracks_from(albums_uri)
+                track_uris_total.extend(tracks_uris)
+            print(track_uris_total)
+            for track_uri in track_uris_total:
+                features_to_db(track_uri)
+
+        elif str_res == 'album':
+            track_uris_total = get_tracks_from(uri)
+            print(track_uris_total)
+            for track_uri in track_uris_total:
+                features_to_db(track_uri)
+
+        elif str_res == 'track':
+            features_to_db(uri)
+
+    except Exception as e:
+        print("Error processing {}: {}".format(uri, e))
+        raise e
+
+    else:
+        DB.session.commit()
+
+
+def features_from_category(cat_id):
+    """Posts tracks from the
+    category to the database
+    :param cat_id: category id
+    :return:
+    """
+
+    try:
+        playlists_uris = get_playlists_from(cat_id)
+        artist_uris_total = []
+        for playlist_uri in playlists_uris:
+            artist_uris = get_artists_from(playlist_uri)
+            artist_uris_total.append(artist for artist in artist_uris)
+        albums_uris_total = []
+        for artist_uri in artist_uris_total:
+            album_uris = get_albums_from(artist_uri)
+            albums_uris_total.append(album for album in album_uris)
+        # from a list of albums get tracks
+        track_uris_total = []
+        for albums_uri in albums_uris_total:
+            tracks_uris = get_tracks_from(albums_uri)
+            track_uris_total.append(track for track in tracks_uris)
+        for track_uri in track_uris_total:
+            features_to_db(track_uri)
+
+    except Exception as e:
+        print("Error processing {}: {}".format(cat_id, e))
+        raise e
```

Comment:
Did you mean to catch the exception, and then raise it again? This will print the Error processing string, and then raise the error again for the next level up to deal with.